gammapy/cube/fit.py

```diff
@@ -208,6 +275,67 @@ def bin_volume(self):
         de = self.energy_bin_width
         return omega * de
 
+    @property
+    def coords(self):
+        """Return evaluator coords"""
+        lon, lat = self.lon_lat
+        if self.edisp:
+            energy = self.edisp.e_reco.nodes[:, np.newaxis, np.newaxis]
+        else:
+            energy = self.energy_center
+        return {"lon": lon.value, "lat": lat.value, "energy": energy}
+
+    @property
+    def needs_update(self):
+        """Check whether the model component has drifted away from its support."""
+        if self.exposure is None:
+            update = True
+        elif isinstance(self.model, SkyModels):
+            update = False
+        else:
+            position = self.model.position
+            separation = self.exposure.geom.center_skydir.separation(position)
+            update = separation > UPDATE_THRESHOLD
+        return update
+
+    def update(self, exposure, psf, edisp, geom):
+        """Update MapEvaluator, based on the current position of the model component.
+
+        Parameters
+        ----------
+        exposure : `Map`
+            Exposure map.
+        psf : `PSFMap`
```

Comment:
At the moment this expects a `PSFKernel` and `EnergyDispersion` right? If we lack time to properly include the `PSFMap` and `EDispMap`I would keep the expected type in the docstring.